### DISCUSSION TITLE
Time dependent wall boundary condition

### DISCUSSION THREAD

**Discussion by rd-contr** [Exawind/amr-wind#398] [Q&A] [answered]
How difficult would it be to implement a time dependent temperature or heat flux for the wall model?

**Comment by michaeljbrazell** [comment, marked as answer]
Currently you can specify a surface temperature that varies linearly with time: 
https://github.com/Exawind/amr-wind/main/amr-wind/wind_energy/ABLWallFunction.cpp#L43-L65

and is updated here:
https://github.com/Exawind/amr-wind/blob/main/amr-wind/wind_energy/ABLWallFunction.cpp#L93-L99

we don't currently have a way to change the flux over time but by changing mo.surf_temp_flux somewhere you have access to time it should be straightforward. check out this chunk of code too:

https://github.com/Exawind/amr-wind/blob/main/amr-wind/wind_energy/MOData.cpp#L53-L63

How do you plan to use this? Change temperature/flux linearly or something non-standard? We have been thinking about adding a table input that would give more control like this, maybe @mchurchf already has something in the works?

**Reply by rd-contr** [reply]
I would be interested in something more non-standard. A table lookup would likely work. We are interested in replicating a GABLS2 test case from Kumar, et al, "Impact of Surface Flux Formulations and Geostrophic Forcing on Large-Eddy Simulations of Diurnal Atmospheric Boundary Layer Flow", _Journal of Applied Meteorology and Climatology_, Vol 49, 2010, pp 1496-1516.

**Reply by michaeljbrazell** [reply]
Oh there are cosines in there! I feel like this could be hacked in easily if you want to take a stab at it. I'll chat with Matt and others to come up with a more permanent plan.

**Reply by rd-contr** [reply]
Thanks for pointing me in the direction of the linear change in the wall function. I should be able to hack something together based on the functions in the article.

**Reply by sayerhs** [reply]
@shashankNREL aren't you working on something similar for your WRF forcing work?
